vpr/src/route/rr_graph_reader.cpp

```diff
@@ -140,45 +141,142 @@ void load_rr_file(const t_graph_type graph_type,
         int max_chan_width = (is_global_graph ? 1 : nodes_per_chan.max);
         VTR_ASSERT(max_chan_width > 0);
 
-        /* Alloc rr nodes and count count nodes */
-        next_component = get_single_child(rr_graph, "rr_nodes", loc_data);
-
-        int num_rr_nodes = count_children(next_component, "node", loc_data);
-
-        device_ctx.rr_nodes.resize(num_rr_nodes);
-        process_nodes(next_component, loc_data);
-
         /* Loads edges, switches, and node look up tables*/
         next_component = get_single_child(rr_graph, "switches", loc_data);
 
         int numSwitches = count_children(next_component, "switch", loc_data);
         device_ctx.rr_switch_inf.resize(numSwitches);
 
         process_switches(next_component, loc_data);
-
-        next_component = get_single_child(rr_graph, "rr_edges", loc_data);
-        process_edges(next_component, loc_data, wire_to_rr_ipin_switch, numSwitches);
-
-        //Partition the rr graph edges for efficient access to configurable/non-configurable
-        //edge subsets. Must be done after RR switches have been allocated
-        partition_rr_graph_edges(device_ctx);
-
-        process_rr_node_indices(grid);
-
-        init_fan_in(device_ctx.rr_nodes, device_ctx.rr_nodes.size());
-
-        //sets the cost index and seg id information
-        next_component = get_single_child(rr_graph, "rr_nodes", loc_data);
-        set_cost_indices(next_component, loc_data, is_global_graph, segment_inf.size());
-
-        alloc_and_load_rr_indexed_data(segment_inf, device_ctx.rr_node_indices,
-                                       max_chan_width, *wire_to_rr_ipin_switch, base_cost_type);
-
-        process_seg_id(next_component, loc_data);
+	/* Branches to binary format */
+	next_component = get_single_child(rr_graph, "binary_nodes_and_edges", loc_data, OPTIONAL);
+	if(next_component) 
+	{
+		auto filename = get_attribute(next_component, "file", loc_data).as_string("");
```

Comment:
Please factor this into a function, rather than inline.

Reply:
I believe I have made both of these changes.  Let me know if I missed anything.